mob

```diff
@@ -249,6 +250,22 @@ if git_commit:
 for pair in build_env:
     docker_run.extend(["--env", pair])
 
+# For many binaries linking takes 2GB per binary. This often happens at the end
+# of a build which can result in all cores trying to link, thus using 2GB per
+# core. For a 16 core machine 32GB would need to be available.
+# This logic will limit the build jobs to prevent out of memory issues.
+# https://github.com/rust-lang/cargo/issues/9157 talks about trying to fix this
+# in cargo.
+jobs = os.environ.get("CARGO_BUILD_JOBS")
+if jobs is None:
+    cpus = psutil.cpu_count()
+    ram = psutil.virtual_memory().total
+    # Remove 5GB for local PC overhead
+    max_jobs = (ram - 5_000_000_000) // 2_000_000_000
+    jobs = min(max_jobs, cpus)
```

Comment:
What if this number is zero or negative?

Reply:
good catch, added logic for that